src/mame/sgi/sgi.cpp

```diff
@@ -506,9 +514,6 @@ void sgi_mc_device::write(offs_t offset, uint32_t data, uint32_t mem_mask)
 	case 0x0028/4:
 		LOGMASKED(LOG_RPSS, "%s: RPSS Divider Write: %08x & %08x\n", machine().describe_context(), data, mem_mask);
 		m_rpss_divider = data;
-		m_rpss_divide_count = (int)(m_rpss_divider & 0xff);
-		m_rpss_divide_counter = m_rpss_divide_count;
-		m_rpss_increment = (m_rpss_divider >> 8) & 0xff;
 		break;
```

Comment:
You need to update the counter at this point before updating the divider.  You’ll get bad results if you don’t (you’ll treat the whole interval as using the new divider).